sktime/forecasting/compose/_networkpipeline.py

```diff
@@ -0,0 +1,398 @@
+            # if estimator has a `predict()` method it is a forecaster
+
+            if hasattr(est, "predict"):
+
+                fitted_estimator = self._fitted_estimators[name]
+                pred = fitted_estimator.predict(
+                    **processed_arguments, return_pred_int=return_pred_int, alpha=alpha
+                )
+                # pred.index.freq = self._y.index.freq
+                self._step_results[name] = pred
+
+        return pred
+
+    def update(self, y, X=None, update_params=True):
+        """Update fitted parameters
+
+        Parameters
+        ----------
+        y : pd.Series
+        X : pd.DataFrame
+        update_params : bool, optional (default=True)
+
+        Returns
+        -------
+        self : an instance of self
+        """
+        self.check_is_fitted()
+        y_index_frequency = self._y.index.freq
+        self._update_y_X(y, X)
+        self._y = self._y.asfreq(y_index_frequency)
+
+        for (name, est, arguments) in self._iter(method="update"):
+            # check arguments of pipeline.
+            # Inspect the `update` key of the step arguments
+            # check and process corresponding values
+
+            # estimator in self.steps_ is not fitted.
+            # use fitted estimator in self._fitted_estimators
+            est = self._fitted_estimators[name]
+
+            if "update" in arguments and arguments["update"] is None:
+                # this step must be skipped
+                continue
+            # assume update takes same arguments as fit
+            elif "update" in arguments:
+
+                processed_arguments = self._process_arguments(arguments["update"])
+
+            else:
+                processed_arguments = self._process_arguments(arguments)
+
+            processed_arguments["update_params"] = update_params
+            # Transformers do not have `predict()`
+
+            if hasattr(est, "update"):
+
+                out = est.update(**processed_arguments)
+                out._cutoff = self.cutoff
+                self._step_results[name] = out
+            else:
+                raise ValueError(f"{name} has no update() method.")
+            #     continue
+            # forecasters have `update()` and `predict()` methods
+            # if hasattr(est, "update") and hasattr(est, "predict"):
+            #     est.update(**processed_arguments)
+
+        self._is_fitted = True
+
+        return self
+
+    def get_params(self, deep=True):
+        """Get parameters for this estimator.
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+
+        return {"steps": self.steps}
```

Comment:
I think this should go through the steps and return their parameters, similar to how a `sklearn` pipeline does it - the steps are not parameters.